### PR TITLE
Tidy presenters, fragments and XML.

### PR DESCRIPTION
**Pull request by jhugman**
Enforce some naming conventions and pairings of presenters and fragments.

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt [hunk 1]
```diff
@@ -14,6 +15,7 @@ interface ListEntriesProtocol {
     // TODO: Item list selection
 }
 
+
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt]: 👾

File: app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt [hunk 2]
```diff
@@ -13,13 +13,13 @@ import mozilla.lockbox.action.RouteAction.*
 import mozilla.lockbox.flux.Presenter
 import mozilla.lockbox.store.RouteStore
 import mozilla.lockbox.view.FxALoginFragment
+import mozilla.lockbox.view.ItemListFragment
 import mozilla.lockbox.view.WelcomeFragment
 
-
-class IntentPresenter(private val activity: AppCompatActivity, routeStore: RouteStore = RouteStore.shared): Presenter() {
-
+class RoutePresenter(private val activity: AppCompatActivity, routeStore: RouteStore = RouteStore.shared): Presenter() {
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt]: nice

File: app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt [hunk 3]
```diff
@@ -1,2 +1,26 @@
...
+    private val compositeDisposable = CompositeDisposable()
+
+    fun onViewReady() {
+        protocol.getStartedClicks.subscribe { _ ->
+            dispatcher.dispatch(RouteAction.LOGIN)
+        }.addTo(compositeDisposable)
+    }
+
+    fun onDestroy() {
+        compositeDisposable.clear()
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt]: no longer necessary?

File: app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt [hunk 4]
```diff
@@ -1,2 +1,26 @@
...
+import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.flux.Dispatcher
+
+interface WelcomeViewProtocol {
+    val getStartedClicks: Observable<Unit>
+}
+
+class WelcomePresenter(private val protocol: WelcomeViewProtocol, val dispatcher: Dispatcher = Dispatcher.shared) {
+
+    private val compositeDisposable = CompositeDisposable()
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt]: should be inherited

File: app/src/main/java/mozilla/lockbox/view/ItemListFragment.kt [hunk 5]
```diff
@@ -32,7 +32,7 @@ class ItemListFragment : Fragment(), ListEntriesProtocol {
     ): View? {
         presenter = ListEntriesPresenter(this)
 
-        val view = inflater.inflate(R.layout.item_list_fragment, container, false)
+        val view = inflater.inflate(R.layout.fragment_item_list, container, false)
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/view/ItemListFragment.kt]: `fragment_this` 👍

File: app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt (hunk 3 above)
Comment (jhugman) [reply app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt]: 👍

File: app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt (hunk 2 above)
Comment (jhugman) [reply app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt]: :+1:

File: app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt (hunk 1 above)
Comment (jhugman) [reply app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt]: ☮️

File: app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt (hunk 4 above)
Comment (jhugman) [reply app/src/main/java/mozilla/lockbox/presenter/WelcomePresenter.kt]: 👍

**Pull request merged**